index.js

```diff
@@ -260,11 +284,17 @@ module.exports = function (schema, options) {
             this.deletedId = params.deletedId
         }
 
-        if (options.validateBeforeDelete === false) {
-            return this.save({ validateBeforeSave: false }, cb);
+        const options = {};
+        if (params.session) {
+            options.session = params.session;
         }
 
-        return this.save(cb);
+        const saveOptions = { 
+            ...options, 
+            validateBeforeSave: params.validateBeforeDelete === true
+        };
```

Comment:
Then here you can do:
```suggestion
        if (options.validateBeforeDelete === false) {
            saveOptions.validateBeforeSave = false
        };
```